### PR TITLE
[FLINK-29165][docs] update docs to create and delete deployment via code

### PR DESCRIPTION
**Pull request by gauravmiglanid11**
## What is the purpose of the change

* update docs to create and delete deployment via code

## Brief change log

  - Updated quickstart.md

## Documentation

  - Does this pull request introduce a new feature? no

### CODE REVIEW

**Comment by mbalassi** [comment]
Hi @gauravmiglanid11,

Thanks for considering this, I do believe that this does add value to the docs. My concern is that it leads to additional maintenance as this code example in the docs will need to be bumped for `FlinkDeployment` and flink docker image versions. Do you see ways to eliminating points of maintenance? The docs does offer some variables that can be set here:
https://github.com/apache/flink-kubernetes-operator/blob/main/docs/config.toml
Maybe you can also get the latest CRD version in the Java code somehow.

**Comment by gauravmiglanid11** [comment]
hi @mbalassi , i didn't got your point, there is only one variable in my changes, i.e. flink version, do you want me to put it in config.toml file, and why `FlinkDeployment`  version bumped is needed in the code example: are you talking about `flink.apache.org/v1beta1`. Please help me so that I can fix my PR

**Comment by morhidi** [comment]
Instead of the docs I guess we could also create an E2E example in the folder: https://github.com/apache/flink-kubernetes-operator/tree/main/examples and simply refer it from the quick-start.

**Comment by gauravmiglanid11** [comment]
that's a good idea, doing it thanks @morhidi

**Comment by gauravmiglanid11** [comment]
Hi @morhidi, should i create another module in examples folder or write code example in flink-sql-runner-example, if yes, can you suggest me the name of module

**Comment by morhidi** [comment]
Please create another module, e.g. `kubernetes-client-examples`

**Comment by gauravmiglanid11** [comment]
@morhidi  code is done, please review and merge

**Comment by morhidi** [comment]
will test it tomorrow, thx

**Comment by gauravmiglanid11** [comment]
cool, @morhidi

File: examples/kubernetes-client-examples/pom.xml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.apache.flink</groupId>
+            <artifactId>flink-kubernetes-operator</artifactId>
+            <version>1.2-SNAPSHOT</version>
+            <scope>compile</scope>
+        </dependency>
+    </dependencies>
+
+</project>
```
Comment (morhidi) [inline examples/kubernetes-client-examples/pom.xml:49]: missing new line

File: examples/kubernetes-client-examples/src/main/java/org/apache/flink/examples/Basic.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import static java.util.Map.entry;
+
+/** client code for ../basic.yaml. */
+public class Basic {
+    public static void main(String[] args) {
+        FlinkDeployment flinkDeployment = new FlinkDeployment();
+        flinkDeployment.setApiVersion("flink.apache.org/v1beta1");
+        flinkDeployment.setKind("FlinkDeployment");
+        ObjectMeta objectMeta = new ObjectMeta();
+        objectMeta.setName("advanced-ingress");
```
Comment (morhidi) [inline examples/kubernetes-client-examples/src/main/java/org/apache/flink/examples/Basic.java:45]: this is not an advanced-ingress example, rather a basic

File: examples/kubernetes-client-examples/src/main/java/org/apache/flink/examples/Basic.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                .getSpec()
+                .setJob(
+                        JobSpec.builder()
+                                .jarURI(
+                                        "local:///opt/flink/examples/streaming/StateMachineExample.jar")
+                                .parallelism(2)
+                                .upgradeMode(UpgradeMode.STATELESS)
+                                .build());
+
+        try (KubernetesClient kubernetesClient =
```
Comment (morhidi) [inline examples/kubernetes-client-examples/src/main/java/org/apache/flink/examples/Basic.java:71]: The example should work as is with minikube, using: KubernetesClient kubernetesClient = new DefaultKubernetesClient(); 

It is fine to add a utility explaining how to connect to a remote Kube API server, but please explain, how to get the token, cert, etc.

File: examples/kubernetes-client-examples/pom.xml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        <artifactId>flink-kubernetes-operator-parent</artifactId>
+        <version>1.2-SNAPSHOT</version>
+        <relativePath>../..</relativePath>
+    </parent>
+
+    <artifactId>kubernetes-client-examples</artifactId>
+    <name>Flink Kubernetes client code Example</name>
+
+    <dependencies>
+        <dependency>
```
Comment (morhidi) [inline examples/kubernetes-client-examples/pom.xml:35]: fabric8 is coming with the operator, no need for this dep here

**Review by morhidi — CHANGES_REQUESTED** [review]
Please add a README.md similarly to other examples explaining what the example does, and how to use it.

File: examples/kubernetes-client-examples/src/main/java/org/apache/flink/examples/Basic.java (hunk 2 above)
Comment (gauravmiglanid11) [reply examples/kubernetes-client-examples/src/main/java/org/apache/flink/examples/Basic.java]: done

File: examples/kubernetes-client-examples/pom.xml (hunk 1 above)
Comment (gauravmiglanid11) [reply examples/kubernetes-client-examples/pom.xml]: added

**Comment by gauravmiglanid11** [comment]
Hi @morhidi , changes done, please review

File: examples/kubernetes-client-examples/README.MD [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+## Overview
+
+This is an end-to-end example of running Flink Job via code using the Flink Kubernetes Operator.
+
+It is only intended to serve as a showcase of how Flink Operator classes can be used to submit flink job the Flink Kubernetes Operator.
+
+
+*What's in this example?*
+
+1. Code approach to the simple basic.yaml
```
Comment (morhidi) [inline examples/kubernetes-client-examples/README.MD:31]: wording: Sample code for submitting an application similar to `examples/basic.yaml` programatically.

File: examples/kubernetes-client-examples/README.MD [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+under the License.
+-->
+
+# Flink Kubernetes Client Code Example
+
+## Overview
+
+This is an end-to-end example of running Flink Job via code using the Flink Kubernetes Operator.
+
+It is only intended to serve as a showcase of how Flink Operator classes can be used to submit flink job the Flink Kubernetes Operator.
```
Comment (morhidi) [inline examples/kubernetes-client-examples/README.MD:26]: nit: Second sentence is probably redundant here

**Comment by morhidi** [comment]
LGTM, some wording suggestions added.

**Comment by gauravmiglanid11** [comment]
Hi @morhidi , changes done, please review

**Comment by morhidi** [comment]
+1 LGTM

**Comment by morhidi** [comment]
cc @gyfora could you please kick the workflow on this?

**Comment by gauravmiglanid11** [comment]
hi @morhidi , all checks passed, can you please merge it

**Pull request merged**